Code/GraphMol/ForceFieldHelpers/CrystalFF/TorsionAngleM6.cpp

```diff
@@ -0,0 +1,150 @@
+// $Id$
+//
+//  Copyright (C) 2015 Sereina Riniker
+//
+//  Copyright (C) 2004-2006 Rational Discovery LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include "TorsionAngleM6.h"
+#include <cmath>
+#include <vector>
+#include <ForceField/ForceField.h>
+#include <RDGeneral/Invariant.h>
+#include <ForceField/MMFF/Params.h>
+#include <ForceField/MMFF/TorsionAngle.h>
+
+namespace ForceFields {
+  namespace CrystalFF {
+    using namespace MMFF;
+
+    double calcTorsionEnergyM6(const std::vector<double> V, const std::vector<int> signs,
+                               const double cosPhi) {
+      double cosPhi2 = cosPhi * cosPhi;
+      double cosPhi3 = cosPhi * cosPhi2;
+      double cosPhi4 = cosPhi * cosPhi3;
+      double cosPhi5 = cosPhi * cosPhi4;
+      double cosPhi6 = cosPhi * cosPhi5;
+
+      double cos2Phi = 2.0 * cosPhi2 - 1.0;
+      double cos3Phi = 4.0 * cosPhi3 - 3.0 * cosPhi;
+      double cos4Phi = 8.0 * cosPhi4 - 8.0 * cosPhi2 + 1.0;
+      double cos5Phi = 16.0 * cosPhi5 - 20.0 * cosPhi3 + 5.0 * cosPhi;
+      double cos6Phi = 32.0 * cosPhi6 - 48.0 * cosPhi4 + 18.0 * cosPhi2 - 1.0;
+
+      return (V[0] * (1.0 + signs[0] * cosPhi)
+              + V[1] * (1.0 + signs[1] * cos2Phi)
+              + V[2] * (1.0 + signs[2] * cos3Phi)
+              + V[3] * (1.0 + signs[3] * cos4Phi)
+              + V[4] * (1.0 + signs[4] * cos5Phi)
+              + V[5] * (1.0 + signs[5] * cos6Phi));
+    }
+
+    TorsionAngleContribM6::TorsionAngleContribM6(ForceFields::ForceField *owner, 
```

Comment:
1. Given that this is constructor, the member variables should be initialized using an initializer list. The ctor itself then just becomes the calls to the preconditions and range checks.
2. the arguments V and signs should almost certainly be passed by reference